instructor/client_watsonx.py

```diff
@@ -0,0 +1,50 @@
+from __future__ import annotations
+
+from typing import overload
+
+from ibm_watsonx_ai.foundation_models import ModelInference as Watsonx
+
+import instructor
+
+ALLOWED_MODES = {instructor.Mode.MD_JSON}
+
+
+@overload
+def from_watsonx(
+    client: Watsonx,
+    mode: instructor.Mode = instructor.Mode.TOOLS,
+    **kwargs,
+) -> instructor.Instructor: ...
+
+
+@overload
+def from_watsonx(
+    client: Watsonx,
+    mode: instructor.Mode = instructor.Mode.TOOLS,
+    **kwargs,
+) -> instructor.AsyncInstructor: ...
+
+
+def from_watsonx(
+    client: Watsonx,
+    mode: instructor.Mode = instructor.Mode.MD_JSON,
+    **kwargs,
+) -> instructor.Instructor | instructor.AsyncInstructor:
+    assert mode in ALLOWED_MODES, f"Mode must be one of {ALLOWED_MODES}"
+
+    assert isinstance(
+        client, Watsonx
+    ), "Client must be an instance of ibm_watsonx_ai.foundation_models.ModelInference"
+
+    # TODO: add watsonx async
+    # if client.async_mode:
+    # return instructor.AsyncInstructor(
+    # else:
+
+    return instructor.Instructor(
+        client=client,
+        create=instructor.patch(create=client.generate_text_stream, mode=mode),
```

Comment:
does this work? it needs to have create(messages=[...])

Reply:
Not yet. I changed the PR to be a draft